consensus/core/src/linearizer.rs

```diff
@@ -34,17 +34,18 @@ impl Linearizer {
         let mut to_commit = Vec::new();
         let mut committed = HashSet::new();
 
-        let timestamp_ms = leader_block.timestamp_ms();
+        let dag_state = self.dag_state.read();
+        let timestamp_ms = leader_block
+            .timestamp_ms()
+            .max(dag_state.last_commit_timestamp_ms());
```

Comment:
Since we are doing this small amendment here I would suggest that we either:
1.  `last_commit_timestamp_ms` as property of the method `collect_sub_dag` , as we already do for the rest details (index, rounds) of the last commit
2. Remove the `last_commit_index` & `last_committed_rounds` from the method and retrieve them as well directly here.

I think that will make things a bit more consistent on the point we retrieve the last commit details